src/irmin-pack/unix/stats.mli

```diff
@@ -0,0 +1,83 @@
+(*
+ * Copyright (c) 2018-2021 Tarides <contact@tarides.com>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *)
+
+module Pack_store : sig
+  type field =
+    | Appended_hashes
+    | Appended_offsets
+    | Staging  (** Found in the store's write buffer. *)
+    | Lru  (** Found in the store's LRU of recent [find] results. *)
+    | Pack_direct
+        (** Decoded directly from the pack file (via a direct key). *)
+    | Pack_indexed
+        (** Binding recovered from the pack file after first checking the index
+            for its offset and length (via an indexed key). *)
+    | Not_found  (** Find returned [None]. *)
+  [@@deriving irmin]
+
+  type t = private {
+    mutable appended_hashes : int;
+    mutable appended_offsets : int;
+    mutable total : int;
+    mutable from_staging : int;
+    mutable from_lru : int;
+    mutable from_pack_direct : int;
+    mutable from_pack_indexed : int;
+  }
+  [@@deriving irmin]
+
+  type stat
+
+  val cache_misses : t -> int
+  val export : stat -> t
+end
+
+module Index : sig
+  type t = Index.Stats.t = private {
+    mutable bytes_read : int;
+    mutable nb_reads : int;
+    mutable bytes_written : int;
+    mutable nb_writes : int;
+    mutable nb_merge : int;
+    mutable merge_durations : float list;
+    mutable nb_replace : int;
+    mutable replace_durations : float list;
+    mutable nb_sync : int;
+    mutable time_sync : float;
+    mutable lru_hits : int;
+    mutable lru_misses : int;
+  }
+  [@@deriving irmin]
+
+  type stat
+
+  val export : stat -> t
+end
+
+type t = { pack_store : Pack_store.stat; index : Index.stat }
+
+val reset_stats : unit -> unit
+val get : unit -> t
+val report_pack_store : field:Pack_store.field -> unit
+val report_index : unit -> unit
```

Comment:
maybe it would be useful to add a comment for the two `report_`? 
to explain that `report_pack_store` [also increments the field](https://github.com/mirage/irmin/pull/1817/files#diff-5b830a017f27442c88e2cb46ac9147c90363357193589954ef831b3a2c9270c5R186) whereas the following `report` function does not; 
also that they are needed before a call to `get`, is that right?

Reply:
You can still get it but it won't be filled with other values than the `initial_state` value.